--- /Users/tad/github/msearch/msearch.cgi	2022-06-30 18:37:58.000000000 +0900
+++ msearch.cgi	2022-06-30 21:32:11.000000000 +0900
@@ -277,6 +277,16 @@
     my $content = $row->{content};
     my $title = $row->{title};
 
+    # ローカルURLに書き換える
+    if ($ENV{HTTP_HOST} ne 'anineco.org') {
+      my $u = $ENV{HTTP_HOST} . $ENV{REQUEST_URI};
+      $u =~ s,/msearch/msearch\.cgi.*$,,;
+      $url =~ s/anineco\.org/$u/;
+      if ($ENV{HTTPS} ne 'on' && $ENV{HTTP_X_FORWARDED_PROTO} ne 'https') {
+        $url =~ s/https:/http:/;
+      }
+    }
+
     my $n = length($content);
     my $i = -1;
     foreach my $w (@c_words) {
